humanitec-resource-defs/redis/basic/main.tf

```diff
@@ -0,0 +1,88 @@
+resource "humanitec_resource_definition" "main" {
+  id          = "${var.prefix}redis"
+  name        = "${var.prefix}redis"
+  type        = "redis"
+  driver_type = "humanitec/template"
+
+  driver_inputs = {
+    values_string = jsonencode({
+      templates = {
+        cookie    = <<EOL
+port: {{ .init.port }}
+EOL
+        init      = <<EOL
+{{- if .cookie}}
+port: {{ .cookie.port }}
+{{- else }}
+port: 6379
+{{- end }}
+name: redis-{{ .id }}
+EOL
+        manifests = <<EOL
+deployment.yaml:
+  location: namespace
+  data:
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: {{ .init.name }}
+    spec:
+      serviceName: mysql
+      replicas: 1
+      selector:
+        matchLabels:
+          app: {{ .init.name }}
+      template:
+        metadata:
+          labels:
+            app: {{ .init.name }}
+        spec:
+          securityContext:
+            fsGroup: 1000
+            runAsGroup: 1000
+            runAsNonRoot: true
+            runAsUser: 1000
+            seccompProfile:
+              type: RuntimeDefault
+          containers:
+          - name: {{ .init.name }}
+            securityContext:
+              allowPrivilegeEscalation: false
+              capabilities:
+                drop:
+                  - ALL
+              privileged: false
+              readOnlyRootFilesystem: true
+            image: redis:alpine
+            ports:
+            - containerPort: {{ .init.port }}
```

Comment:
This won't work as redis always runs on `6379` unless changed by config. I think the easiest would be hardcoding `6379` here and modifying the service port mapping below to:

```
      ports:
      - name: tcp-redis
        port: {{ .init.port }}
        targetPort: `6379`
```

This makes the "public" facing port customisable, but keeps the internal port constant.